src/js/messaging/containers/Main.jsx

```diff
@@ -35,13 +41,66 @@ class Main extends React.Component {
     this.props.setNewFolderName(field);
   }
 
-  handleSubmit() {}
+  handleCreateNewFolderModal() {
+    // Reset the form / state object when closed
+    const resetForm = { value: '', dirty: false };
+    this.props.setNewFolderName(resetForm);
+    this.props.toggleCreateFolderModal();
+  }
+
+  handleSubmit(domEvent) {
+    domEvent.preventDefault();
+    const input = domEvent.target.getElementsByTagName('input')[0];
+    // If, by chance, the veteran has submitted this form without touching the
+    // folder name field, trigger an action that will trigger an error.
+    if (input.value === '') {
+      this.props.setNewFolderName({ value: '', dirty: true });
+    } else {
+      this.props.createNewFolder(input.value);
+    }
+  }
+
+  validateFolderName(existingFolders = [], folderName, dirty = false) {
```

Comment:
I think the validation should've stayed in the modal class. It would still work by passing the available folders as a prop to the modal.